adventure/adventure.py

```diff
@@ -4161,20 +4259,60 @@ def E(t: str) -> str:
             types = [" normal", " rare", "n epic", " legendary"]
             chest_type = types[special.index(1)]
             phrase += (
-                f"\n{bold(to_reward)} {word} been awarded {xp} xp and found {cp} {currency_name}. "
+                f"\n{bold(to_reward)} {word} been awarded {xp_bonus} xp and found {cp_bonus} {currency_name}. "
                 f"You also secured **a{chest_type} treasure chest**!"
             )
         elif special is not False and sum(special) > 1:
             phrase += (
-                f"\n{bold(to_reward)} {word} been awarded {xp} xp and found {cp} {currency_name}. "
+                f"\n{bold(to_reward)} {word} been awarded {xp_bonus} xp and found {cp_bonus} {currency_name}. "
                 f"You also secured **several treasure chests**!"
             )
         else:
             phrase += (
-                f"\n{bold(to_reward)} {word} been awarded {xp} xp and found {cp} {currency_name}."
+                f"\n{bold(to_reward)} {word} been awarded {xp_bonus} xp and found {cp_bonus} {currency_name}."
             )
         return phrase
 
+    async def _reward_bonus(self, ctx, userlist, xp, cp):
+        phrase = ""
+        xp_bonus = 0
+        cp_bonus = 0
+        xp_bonus_list_names = []
+        cp_bonus_list_names = []
+        for user in userlist:
+            try:
+                c = await Character._from_json(self.config, user)
+            except Exception:
+                log.error("Error with the new character sheet", exc_info=True)
+                return
+            if c.race == "dwarf" or c.race == "human":
+                roll = random.randint(1, 100)
+                chance = max(1, int(c.lvl * 25 / 50))
+                if roll in range(1, chance):
+                    bonus = int(1.5 * chance)
+                    if c.race == "dwarf":
+                        cp_bonus += bonus
+                        cp_bonus_list_names.append(self.E(user.display_name))
+                    else:
+                        xp_bonus += bonus
+                        xp_bonus_list_names.append(self.E(user.display_name))
+        if len(xp_bonus_list_names) > 0:
+            bonus = max(1, int(xp * min(100, xp_bonus) / 100))
+            xp += bonus        
+            phrase += (
+                f"\n{bold(humanize_list(xp_bonus_list_names))} led this battle with great panache... "
+                f"*[+{bonus * len(userlist)} xp for the party!]*"
```

Comment:
This is going to make it seem like a huge bonus... if you have like 10 users and 200 xp then it'll say "[+2000xp for the party]", which is technically true... but best just to say "[+200xp each]" ?